external-crates/move/crates/move-compiler/src/naming/ast.rs

```diff
@@ -893,46 +903,46 @@ impl Type_ {
 
     pub fn multiple_(loc: Loc, mut tys: Vec<Type>) -> Type_ {
         match tys.len() {
-            0 => Type_::Unit,
+            0 => UNIT_TYPE.clone(),
             1 => tys.pop().unwrap().value,
-            n => Type_::Apply(None, sp(loc, TypeName_::Multiple(n)), tys),
+            n => TypeInner::Apply(None, sp(loc, TypeName_::Multiple(n)), tys).into(),
         }
     }
 
     pub fn builtin_name(&self) -> Option<&BuiltinTypeName> {
-        match self {
-            Type_::Apply(_, sp!(_, TypeName_::Builtin(b)), _) => Some(b),
+        match &*self.0 {
```

Comment:
All the `&* .0` is fine by me, but we could do some borrow/deref trait shenanigans if we want

Reply:
I added the `inner` helper for that purpose, because derefing can make cloning harder later otherwise -- Rust's ergonomics here just make things a bit wonky. Hopefully the isolation in this file is acceptable.